packages/nodes-base/nodes/Okta/GenericFunctions.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+	IDataObject,
+	IExecuteFunctions,
+	IExecuteSingleFunctions,
+	IHookFunctions,
+	IHttpRequestMethods,
+	IHttpRequestOptions,
+	ILoadOptionsFunctions,
+	INodeListSearchResult,
+	INodePropertyOptions,
+} from 'n8n-workflow';
+
+type OktaUser = {
+	profile: {
+		firstName: string;
+		lastName: string;
+	};
+	id: string;
+};
+
+export async function oktaApiRequest(
+	this: IExecuteFunctions | IExecuteSingleFunctions | IHookFunctions | ILoadOptionsFunctions,
+	method: IHttpRequestMethods,
+	resource: string,
+	body: IDataObject = {},
+	qs: IDataObject = {},
+	url?: string,
+	option: IDataObject = {},
+): Promise<OktaUser[]> {
+	const credentials = await this.getCredentials('oktaApi');
+	const baseUrl = `${credentials.url as string}/api/v1/${resource}`;
+	const options: IHttpRequestOptions = {
+		headers: {
+			'Content-Type': 'application/json',
+		},
+		method,
+		body: Object.keys(body).length ? body : undefined,
+		qs: Object.keys(qs).length ? qs : undefined,
+		url: url ?? baseUrl,
+		json: true,
+		...option,
+	};
+	return await (this.helpers.httpRequestWithAuthentication.call(
+		this,
+		'oktaApi',
+		options,
+	) as Promise<OktaUser[]>);
+}
+
+export async function getUsers(
+	this: ILoadOptionsFunctions,
+	filter?: string,
+): Promise<INodeListSearchResult> {
+	const responseData: OktaUser[] = await oktaApiRequest.call(this, 'GET', '/users/');
```

Comment:
I don't think this will get all users after the default 200 because `oktaApiRequest` doesn't implement pagination

Reply:
[emptyArrayItems](https://developer.okta.com/docs/reference/api/users/#list-all-users)
Returns a list of all users that do not have a status of DEPROVISIONED, up to the maximum (200 for most orgs)
not sure we can control this one

Reply:
I added cursor pagination to handle this one